### PR TITLE
v0.27: New instance options

### LINKED ISSUE #1581: v0.27: New instance options `--max-indexing-memory` and `--max-indexing-threads`
These two new instance options give users some influence over resource usage when indexing.

_Important: --max-indexing-memory is not completely reliable and has some margin of error—meaning, an instance with --max-indexing-memory=2gb still go a bit over that limit. We don't have enough data to say how big the error margin is._

## Tasks
- [x] add `--max-indexing-memory` and `--max-indexing-threads` to configuration overview (under `/learn/configuration`
- [x] mention these two new options on "Indexation and performance"
- [ ] also mention options on the FAQ

## References
- https://github.com/meilisearch/meilisearch/issues/2212
- https://github.com/meilisearch/specifications/pull/119
- SME: @curquiza

### PR DESCRIPTION
**Pull request by guimachiavelli**
This PR updates `/learn/configuration/instance_options.md` and `/learn/advanced/indexation.md` with two new instance options: `--max-indexing-memory` and `--max-indexing-threads`.

Resolves #1581

### Code samples updated

None.

### CODE REVIEW

File: learn/configuration/instance_options.md [hunk 1]
```diff
@@ -205,6 +205,40 @@ The `index` stores processed data and is different from the `task` database, whi
...
+### Max indexing threads
+
+**Environment variable**: `MEILI_MAX_INDEXING_THREADS`
+**CLI option**: `--max-indexing-threads`
+**Default value**: half of the available threads
+**Expected value**: an integer
+
+Sets the maximum number of threads Meilisearch can use during indexation. By default, the indexer avoids using more than half of a machine's total processing units. This ensures Meilisearch is always ready to perform searches, even while you are updating an index.
+
+If the `--max-indexing-threads` is higher than the real number of core available in the machine, Meilisearch uses the maximum number of available cores.
```
Comment (maryamsulemani97) [inline learn/configuration/instance_options.md:234]:
```suggestion
If the `--max-indexing-threads` is higher than the real number of cores available in the machine, Meilisearch uses the maximum number of available cores.
```

**Review by maryamsulemani97 — COMMENTED** [review]
Will this require changes in `/faq.md#search-speed`?

**Review by bidoubiwa — APPROVED** [review]
🔥 🔥 LGTM 🔥 🔥  

Very clear and easy to understand

**Review by curquiza — APPROVED** [review]
LGTM :)

**Comment by brunoocasali** [comment]
@guimachiavelli, I think the label `code sample changes needed` is unnecessary here!

File: learn/configuration/instance_options.md [hunk 2]
```diff
@@ -205,6 +205,40 @@ The `index` stores processed data and is different from the `task` database, whi
...
+### Max indexing threads
+
+**Environment variable**: `MEILI_MAX_INDEXING_THREADS`
+**CLI option**: `--max-indexing-threads`
+**Default value**: half of the available threads
+**Expected value**: an integer
+
+Sets the maximum number of threads Meilisearch can use during indexation. By default, the indexer avoids using more than half of a machine's total processing units. This ensures Meilisearch is always ready to perform searches, even while you are updating an index.
+
+If the `--max-indexing-threads` is higher than the real number of cores available in the machine, Meilisearch uses the maximum number of available cores.
```
Comment (maryamsulemani97) [inline learn/configuration/instance_options.md:234]:
```suggestion
If `--max-indexing-threads` is higher than the real number of cores available in the machine, Meilisearch uses the maximum number of available cores.
```

**Review by maryamsulemani97 — APPROVED** [review]
Just a minor change 🧞‍♂️

File: learn/advanced/indexation.md [hunk 3]
```diff
@@ -4,22 +4,26 @@ Adding new documents to an index is a multi-threaded and memory-intensive operat
...
+
+2. Lower-end machines might struggle when processing huge datasets. Splitting your data payload into smaller batches can help in this case. [For more information, consult the section below](#memory-crashes)
 
 ## Multi-threading
 
 In machines with multi-core processors, the indexer avoids using more than half of the available processing units. For example, if your machine has twelve cores, the indexer will try to use six of them at most. This ensures Meilisearch is always ready to perform searches, even while you are updating an index.
 
+You can override Meilisearch's default threading limit by using the `--max-indexing-threads` instance option. Allowing Meilisearch to use all processor cores for indexation might negatively impact your users' search experience.
```
Comment (dichotommy) [inline learn/advanced/indexation.md:25]:
```suggestion
You can override Meilisearch's default threading limit by using the `[--max-indexing-threads](/learn/configuration/instance_options.md#max-indexing-threads)` instance option. Allowing Meilisearch to use all processor cores for indexation might negatively impact your users' search experience.
```

**Comment by guimachiavelli** [comment]
bors merge

**Pull request merged**